gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -45,6 +45,30 @@ struct PatternBinding
   {}
 };
 
+// Info that gets stored in the map. Helps us detect if two bindings to the same
+// identifier have different mutability or ref states.
+struct BindingInfo
```

Comment:
I think BindingInfo is too general and it sounds like it should contain the pattern or identifier. I would call this: BindingTypeInfo